packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -389,21 +399,86 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 
 		if (!executions.length) {
 			if (deleteConditions.ids) {
-				LoggerProxy.error('Failed to delete an execution due to insufficient permissions', {
+				Logger.error('Failed to delete an execution due to insufficient permissions', {
 					executionIds: deleteConditions.ids,
 				});
 			}
 			return;
 		}
 
 		const executionIds = executions.map(({ id }) => id);
-		const binaryDataManager = BinaryDataManager.getInstance();
-		await binaryDataManager.deleteBinaryDataByExecutionIds(executionIds);
-
 		do {
 			// Delete in batches to avoid "SQLITE_ERROR: Expression tree is too large (maximum depth 1000)" error
-			const batch = executionIds.splice(0, 500);
-			await this.delete(batch);
+			const batch = executionIds.splice(0, PRUNING_BATCH_SIZE);
+			await this.softDeleteExecutions(batch);
 		} while (executionIds.length > 0);
 	}
+
+	private async pruneOlderExecutions() {
```

Comment:
Since `older` is redundant and `prune` is not exactly what happens here, maybe `softDeleteExecutionsToPrune()`? Assuming pruning still means hard deletion which is what it has meant so far.